### PR TITLE
feat(secrets): add secrets component

### PR DESCRIPTION
**Pull request by avi-alpert**
Do not merge until:

- [ ] Secrets are supported in the wizard

### Issue

Issue number, if available, prefixed with "#"

### Description

What does this implement? Explain your changes.
-Adds Secret component and SecretDefinition for wizard Options
- fix typo in test blueprint ("touple" -> "tuple")

### Testing

How was this change tested?

### Additional context

Add any other context about the PR here.

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by avi-alpert** [comment]
todo: fix build errors, and fix package.json changes from * to hardcoded versions

File: packages/blueprints/test-blueprint/src/blueprint.ts [hunk 1]
```diff
@@ -60,31 +61,31 @@ export interface Options extends ParentOptions {
   };
 
   /**
-   * These are touple parings.
+   * These are tuple parings.
    * @collapsed
    */
-  toupleValues?: {
+  tupleValues?: {
```
Comment (alexforsyth) [inline packages/blueprints/test-blueprint/src/blueprint.ts:67]: can e pull all this tuple stuff into its own PR?

File: packages/components/secrets/src/secret-definition.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export type SecretDefinition = {
```
Comment (alexforsyth) [inline packages/components/secrets/src/secret-definition.ts:1]: It may make sense to have this be JSON annotated

File: packages/components/secrets/package.json [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    "upgrade": "npx projen upgrade",
+    "watch": "npx projen watch",
+    "projen": "npx projen --no-post",
+    "component:package": "yarn build && yarn package",
+    "npm:push": "yarn bump && yarn component:package && yarn npm:publish",
+    "npm:publish": "npm publish dist/js/*.tgz"
+  },
+  "devDependencies": {
+    "@amazon-codecatalyst/blueprint-util.projen-blueprint-component": "*",
+    "@amazon-codecatalyst/blueprints.blueprint": "*",
```
Comment (alexforsyth) [inline packages/components/secrets/package.json:27]: Is this a dev dependency or just a straight up dependency?

File: packages/components/secrets/package.json [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  "peerDependencies": {
+    "@amazon-codecatalyst/blueprints.blueprint": "*",
+    "projen": "^0.71.112"
+  },
+  "dependencies": {
+    "projen": "0.71.112"
+  },
+  "main": "lib/index.js",
+  "license": "Apache-2.0",
+  "version": "0.0.0",
```
Comment (alexforsyth) [inline packages/components/secrets/package.json:50]: Version should be set to the same increment as the rest of the components

File: packages/components/secrets/package.json [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    "eslint-import-resolver-node": "*",
+    "eslint-import-resolver-typescript": "*",
+    "eslint-plugin-import": "*",
+    "npm-check-updates": "^16",
+    "projen": "*",
+    "standard-version": "^9",
+    "ts-node": "^10",
+    "typescript": "^5.3.3"
+  },
+  "peerDependencies": {
```
Comment (alexforsyth) [inline packages/components/secrets/package.json:41]: I dont think you need this explictly listed as a peer. 
```
@amazon-codecatalyst/blueprints.blueprint"
```

File: packages/components/secrets/src/secret-component.ts [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+      }
+
+      if (this.description && this.description.length > 1000) {
+        blueprint.throwSynthesisError({
+          name: BlueprintSynthesisErrorTypes.ValidationError,
+          message: 'Secret description must have a maxmimum of 1000 characters.',
+        });
+      }
+
+      const shortName = this.name.slice(0, 5);
```
Comment (alexforsyth) [inline packages/components/secrets/src/secret-component.ts:57]: Does this work if the secret is like 4 characters long? 

Probably. just curious. Feel free to resolve

File: packages/blueprints/test-blueprint/src/blueprint.ts [hunk 7]
```diff
@@ -252,6 +253,8 @@ export interface Options extends ParentOptions {
    * @validationRegex /^[A-Z][A-Za-z]{4,24}$/
    */
   pascalCaseRegex: string;
+
+  secret: SecretDefinition;
```
Comment (alexforsyth) [inline packages/blueprints/test-blueprint/src/blueprint.ts:257]: This is good. You should keep this in this PR

**Review by alexforsyth — CHANGES_REQUESTED** [review]
PR looks great. Some minor points to consider.

Please address feedback. Ping me and i'll +1.

File: packages/components/secrets/src/secret-component.ts (hunk 6 above)
Comment (avi-alpert) [reply packages/components/secrets/src/secret-component.ts:57]: yes it does

File: packages/components/secrets/package.json (hunk 3 above)
Comment (avi-alpert) [reply packages/components/secrets/package.json]: fixed

File: packages/components/secrets/package.json (hunk 5 above)
Comment (avi-alpert) [reply packages/components/secrets/package.json:40]: fixed

**Review by alexforsyth — APPROVED** [review]

**Comment by avi-alpert** [comment]
add npm public in package json: https://github.com/aws/codecatalyst-blueprints/pull/479/files
